src/scene_graph/camera_node.hpp

```diff
@@ -9,20 +10,49 @@ namespace wr
 
 	struct CameraNode : Node
 	{
+		using FovDefault = util::NamedType<float>;
+		using FovFocalLength = util::NamedType<float>;
+		using FovAspectRatio = util::NamedType<float>;
+		using FovFilmSize = util::NamedType<float>;
+
+		struct FoV
+		{
+			FoV(FovDefault deg) : m_fov(deg.Get() / 180.0f * 3.1415926535f)
+			{
+			}
+
+			FoV(FovFocalLength focal_length, FovAspectRatio aspect_ratio, FovFilmSize film_size) : 
+				m_fov(2.0f * std::atan2(film_size.Get() / aspect_ratio.Get(), 2.0f * focal_length.Get()))
+			{
+			}
+
+			float m_fov;
+		};
+
 		CameraNode(float fov_deg, float aspect_ratio)
 			: Node(),
 			m_active(true),
 			m_frustum_near(0.1f),
 			m_frustum_far(1000.0f),
-			m_fov(fov_deg / 180 * 3.1415926535f),
-			m_aspect_ratio(aspect_ratio)
+			m_aspect_ratio(aspect_ratio),
+			m_focal_length(35.0f),
+			m_film_size(45.0f),
+			m_fov(FovFocalLength(m_focal_length),FovAspectRatio(aspect_ratio), FovFilmSize(m_film_size)),
```

Comment:
We want 2 constructors since the user might want the different constructor when writing code.